api/admin/validator.py

```diff
@@ -163,3 +165,36 @@ def _value(self, field, form):
         elif len(value) == 1:
             return value[0]
         return filter(lambda x: x != None and x != "", value)
+
+
+class ValidatorFactory(object):
+    """Creates Validator instances for particular authentication protocols"""
+
+    def __init__(self):
+        """Initializes a new instance of ValidatorFactory class"""
+        self._logger = logging.getLogger(__name__)
+
+    def create(self, protocol):
+        """
+        Returns a configured validator for a specific protocol
+
+        :param protocol: Patron authentication protocol
+        :type protocol: string
+
+        :return: Validator object (if there is any for the specific protocol)
+        :rtype: Optional[Validator]
+        """
+        module_name = protocol
+
+        try:
+            provider_module = importlib.import_module(module_name)
+            validator_class = getattr(provider_module, 'Validator', None)
```

Comment:
(The part that's specific to patron auth protocols is this bit -- we don't do this importlib trick for other types of integrations.)